samples/net/openthread/coap_client/src/coap_utils.c

```diff
@@ -0,0 +1,268 @@
+#include "coap_utils.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(coap_utils, CONFIG_COAP_UTILS_MOD_LOG_LEVEL);
+
+#include <errno.h>
+#include <sys/printk.h>
+#include <sys/byteorder.h>
+#include <zephyr.h>
+
+#include <net/socket.h>
+#include <net/net_mgmt.h>
+#include <net/net_ip.h>
+#include <net/udp.h>
+#include <net/coap.h>
+
+#define PEER_PORT 5683
+#define MAX_COAP_MSG_LEN 256
+#define COAP_VER 1
+#define COAP_TOKEN_LEN 8
+#define COAP_MAX_REPLIES 1
+#define COAP_POOL_SLEEP 500
+#define COAP_RECEIVE_STACK_SIZE 500
+
+static K_THREAD_STACK_DEFINE(receive_stack_area, COAP_RECEIVE_STACK_SIZE);
+static struct k_thread receive_thread_data;
+
+const static int nfds = 1;
+static struct pollfd fds;
+static struct coap_reply replies[COAP_MAX_REPLIES];
+
+static void coap_open_socket(int *sock) {
+	while (1) {
+		*sock = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+		if (*sock < 0) {
+			LOG_ERR("Failed to create UDP socket %d", errno);
```

Comment:
I'd add a `k_sleep` here, otherwise, you'll starve all lower prio threads with this busy loop if the problem persists (for instance logger runs at the lowest priority).

In general, I wouldn't expect a socket creation failure in such a simple app, unless someone starts playing with the configuration or some bug is introduced in the networking stack. But in case this happens, it would be easier to spot if the looger thread is allowed to run.